File: H3/ScumBoard/ScrumBoardAPI/ScrumBoardAPI/Models/Post.cs
namespace ScrumBoardAPI.Models
{
    public class Post
    {
        public Post(int id, string title, List<Note> notes)
        {
            _id = id;
            _title = title;
            _notes = notes;
        }

        public int Id { get => _id; set => _id = value; }
        public string Title { get => _title; set => _title = value; }
        public List<Note> Notes { get => _notes; set => _notes = value; }

        private int _id;
        private string _title;
        private List<Note> _notes;
    }
}

File: H3/ScumBoard/scrumboard/lib/services/models/post.dart
import 'package:scrumboard/services/models/note.dart';

class Post {
  Post({
    required this.id,
    required this.title,
    required this.notes,
  });

  final int id;
  final String title;
  final List<Note> notes;

  factory Post.fromJson(Map<String, dynamic> json) => Post(
        id: json["Id"],
        title: json["Title"],
        notes: List<Note>.from(json["Notes"].map((x) => Note.fromJson(x))),
      );

  Map<String, dynamic> toJson() => {
        "Id": id,
        "Title": title,
        "Notes": List<dynamic>.from(notes.map((x) => x.toJson())),
      };
}
